### PR TITLE
query by `from` and `until`

### PR DESCRIPTION
**Pull request by ch40s3k**

### CODE REVIEW

File: alamo/api/views.py [hunk 1]
```diff
@@ -40,11 +41,32 @@ def get_queryset(self):
...
+        until = self.request.query_params.get('until', None)
 
-        return CheckResults.objects.filter(
+        queryset = CheckResults.objects.filter(
             check_uuid=check_uuid,
             trigger_uuid__in=triggers,
             insertion_time__gt=now - timedelta(hours=duration)
         )
+
+        if from_date or until:
```
Comment (pnuckowski) [inline alamo/api/views.py]: make it a big shorter  :) 
```
queryset = CheckResults.objects.filter(check_uuid=check_uuid,trigger_uuid__in=triggers)
if from_date or until:
  # ...
  queryset = queryset.filter(insertion_time__gt=from_date, insertion_time__lte=until)
else:
  now = timezone.now()
  queryset = queryset.filter(insertion_time__gt=now - timedelta(hours=duration))
return queryset
```

**Comment by h00dy** [comment]
👊

**Comment by pawellesniewski** [comment]
:imp:

File: alamo/api/views.py [hunk 2]
```diff
@@ -40,11 +41,32 @@ def get_queryset(self):
...
+
+            try:
+                until = parser.parse(until)
+            except AttributeError:
+                until = timezone.now()
+            try:
+                from_date = parser.parse(from_date)
+            except AttributeError:
```
Comment (pnuckowski) [inline alamo/api/views.py]: according to doc https://dateutil.readthedocs.org/en/latest/parser.html#dateutil.parser.parse
`parse` method could raise `ValueError` or `OverflowError `. Why AttributeError?

Comment (ch40s3k) [inline alamo/api/views.py]: Exception block is for the situation if values are None

**Pull request merged**